ucca/core.py

```diff
@@ -248,6 +288,15 @@ def root(self):
     def parent(self):
         return self._parent
 
+    @property
+    def categories(self):
+        return self._categories
+
+    @categories.setter
+    def categories(self, new_categories):
```

Comment:
Since it's a setter, I would expect it to *replace* the list of categories rather than adding to it